### PR TITLE
[MAINT] Stability improvements to stopping measurements in FtBuffer and MNE Scan

### PR DESCRIPTION
**Pull request by gabrielbmotta**
- Add outs for loops where fetching data could get stuck.
- Change order of function calls when ending measurements.

### CODE REVIEW

**Review by juangpc — DISMISSED** [review]

**Review by LorenzE — DISMISSED** [review]
Let's discuss this in our meeting tomorrow. This stuff enters the "danger zone" ;)

File: applications/mne_scan/plugins/ftbuffer/ftbuffer.cpp [hunk 1]
```diff
@@ -146,6 +146,9 @@ bool FtBuffer::stop()
 
     m_bIsConfigured = false;
 
+    requestInterruption();
+    wait(500);
```
Comment (LorenzE) [inline applications/mne_scan/plugins/ftbuffer/ftbuffer.cpp:150]: Can you try if this also works without the 500ms wait?

Comment (gabrielbmotta) [reply applications/mne_scan/plugins/ftbuffer/ftbuffer.cpp:150]: It seems to work fine without the wait command. Escaping the main run loop is pretty fast, so maybe it doesn't affect it with what we are doing now. The [wait call](https://doc.qt.io/qt-5/qthread.html#wait) does stop blocking early if the function returns quickly (which it seems to).

**Comment by juangpc** [comment]
What state this is in? ready for merging? 
I have a couple of questions. Perhaps during the day.

**Review by juangpc — APPROVED** [review]

**Pull request merged**